auto-self-nominate/src/AccountState.ts

```diff
@@ -0,0 +1,100 @@
+import { SDK } from "codechain-sdk";
+import { PlatformAddress } from "codechain-sdk/lib/core/classes";
+
+const RLP = require("rlp");
+
+export type AccountState = "Banned" | "Candidates" | "Jailed";
+export interface JailedAccountInfo {
+    deposits: number;
+    custodyUntil: number;
+    releasedAt: number;
+}
+export interface CandidatesInfo {
+    [key: string]: {
+        deposits: number;
+        nominateEndAt: number;
+    };
+}
+export interface BannedAccountsInfo {
+    accounts: string[];
+}
+export interface JailedAccountsInfo {
+    [key: string]: {
+        deposits: number;
+        custodyUntil: number;
+        releasedAt: number;
+    };
+}
+
+export type StatesInfos = CandidatesInfo | BannedAccountsInfo | JailedAccountsInfo;
+
+function decodeNumber(buffer: Buffer): number {
+    return parseInt(buffer.toString("hex"), 16);
+}
+
+function decodePlatformAddressFromPubkey(networkId: string, buffer: Buffer): PlatformAddress {
+    const pubkey = buffer.toString("hex");
+    return PlatformAddress.fromPublic(pubkey, {
+        networkId,
+    });
+}
+
+function decodePlatformAddressFromAccountId(networkId: string, buffer: Buffer): PlatformAddress {
+    const accountId = buffer.toString("hex");
+    return PlatformAddress.fromAccountId(accountId, {
+        networkId,
+    });
+}
+
+function bannedAccountsDecoder(networkId: string, data: string): BannedAccountsInfo {
+    const list: Buffer[] = RLP.decode(Buffer.from(data, "hex"));
+    return { accounts: list.map(id => decodePlatformAddressFromAccountId(networkId, id).value) };
```

Comment:
Prefer using toString method to value property.

Reply:
I changed all similar routines.